vcsclient/gitlab.go

```diff
@@ -23,6 +23,24 @@ type GitLabClient struct {
 	logger   Log
 }
 
+// GetCommitStatus on GitLab
+func (client *GitLabClient) GetCommitStatus(ctx context.Context, owner, repository, ref string) (status []CommitStatus, err error) {
+	statuses, _, err := client.glClient.Commits.GetCommitStatuses(repository, ref, nil)
```

Comment:
Let's pass the context whenever possible
```suggestion
	statuses, _, err := client.glClient.Commits.GetCommitStatuses(repository, ref, nil, gitlab.WithContext(ctx))
```